### ISSUE TITLE
support for --dry-run option.

### ISSUE THREAD

**Issue by jonahwilliams** [flutter/flutter#43131]
Requirements:
  - Should only be accessible to users with commit access to flutter, and severely rate limited. Otherwise it may excessively compete with the regular process.
  - Should display results in a similar table to the existing devicelab process
  - Must not effect regular build process.

Per discussion with @yjbanov, the Devicelab's Checklist entity encodes the repository path and commit has in the key. With some support added to the agent process, we can teach it to checkout other repositories and run them. 

We'll need to ensure that these results can be communicated without effecting the overall build status. Currently, the dashboard may just be looking at the last checklist, we'll need to force it to look at the last flutter/flutter checklist.

Finally, to ensure that the process has limited availability:

we can create a command script in dev which opens a PR to a private flutter/dry_run repo. This repo contains a single file with a repository and hash that the devicelab will use to create a dry run checklist. We can teach the backend process to listen to changes on this repo. When a dry run is pending a secondary "lock" file can be committed or something similar. The script could attempt to auto merge if the repo is not locked.

**Comment by godofredoc** [comment]
@jonahwilliams would running tests on presubmit solve this problem?

**Comment by jonahwilliams** [comment]
Yes - the more tests are run on presubmit, the less useful this option us
